resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -905,6 +905,10 @@ private[spark] class Client(
       env(ENV_DIST_CLASSPATH) = dcp
     }
 
+    sys.env.get(SPARK_TESTING).foreach { testing =>
+      env(SPARK_TESTING) = testing
+    }
+
```

Comment:
Maybe combine these two?

```suggestion
    Seq(ENV_DIST_CLASSPATH, SPARK_TESTING).foreach { envVar =>
      sys.env.get(envVar).foreach(value => env(envVar) = value)
    }
```

Reply:
looks good

Reply:
59a618e fix this